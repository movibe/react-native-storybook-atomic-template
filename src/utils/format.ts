export const formatPoints = (points: string | number): string => {
  const minus = Number(points) < 0

  let n = Number(points).toString()
  let r = ''
  let x = 0

  for (let i = n.length; i > 0; i--) {
    r += n.substr(i - 1, 1) + (x == 2 && i != 1 ? '.' : '')
    x = x == 2 ? 0 : x + 1
  }
  return `${minus ? '-' : ''}${r.split('').reverse().join('')}`
}

export interface IFlightsStops {
  name: string
  iata: string
  duration: string
}
