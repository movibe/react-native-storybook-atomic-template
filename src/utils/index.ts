export * from './faker'
export * from './qualityAssurance'
export * from './format'
export * from './withTheme'
