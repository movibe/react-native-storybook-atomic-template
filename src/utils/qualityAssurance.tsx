import { Platform } from 'react-native'

interface ItestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}
const testID = (testID: string): ItestID =>
  Platform.OS === 'android'
    ? {
        testID,
        accessible: true,
        accessibilityLabel: testID,
      }
    : {
        testID,
        accessible: false,
      }

type componentType =
  | 'ActivityIndicator'
  | 'View'
  | 'Lottie'
  | 'ScrollView'
  | 'Modal'
  | 'TextInput'
  | 'Image'
  | 'Text'
  | 'ImageBackground'
  | 'TouchableOpacity'
  | 'Radio'
  | 'Checkbox'

export const assignTestId = (componentType: componentType, componentId: string): ItestID => {
  const id = `${componentType}_${componentId}`
  return testID(id)
}
export const mockTestID = (componentType: componentType, componentId: string): ItestID => assignTestId(componentType, componentId)
