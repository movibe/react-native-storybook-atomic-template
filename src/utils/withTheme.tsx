import { boolean, radios } from '@storybook/addon-knobs'
import * as React from 'react'
import styled, { DefaultTheme, ThemeProvider } from 'styled-components/native'

import { Column } from '../atomic/atoms/Flex'
import { themes } from '../tokens'

const options = {
  Light: 'light',
  Dark: 'dark',
}

interface IProps {
  children: any
  padding?: boolean
}

const Background = styled.View`
  background-color: ${({ theme }) => theme.colors.background};
  height: 100vh;
  flex: 1;
`
export const StorybookHoc: React.FC<IProps> = ({ children, padding = true }) => {
  const color = radios('theme_color', options, 'light')
  const _theme: DefaultTheme = color === 'dark' ? themes.dark : themes.light

  return (
    <ThemeProvider theme={_theme}>
      <Background>
        <Column content={boolean('theme_padding', !!padding)} flex={1}>
          {children}
        </Column>
      </Background>
    </ThemeProvider>
  )
}

export const withTheme = (fn: any) => <StorybookHoc>{fn()}</StorybookHoc>
