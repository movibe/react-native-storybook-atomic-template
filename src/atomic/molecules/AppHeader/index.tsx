import * as React from 'react'
import { TouchableOpacity } from 'react-native'

import logoImage from '../../../assets/images/logo.svg'
import { Column } from '../../../atomic/atoms/Flex'
import { assignTestId } from '../../../utils'
import * as S from './styles'

export interface IAppHeaderProps {
  testID?: string
  title?: string
  subtitle?: string
  logo?: boolean
  onBack?: () => void
  onFilter?: () => void
}

export const AppHeader: React.FC<IAppHeaderProps> = ({ testID = 'AppHeader', title, subtitle, logo, onBack, onFilter }): JSX.Element => {
  return (
    <S.Container testID={testID}>
      <S.ContainerTop alignItems='center'>
        <S.Left>
          {onBack && (
            <TouchableOpacity onPress={onBack} {...assignTestId('TouchableOpacity', `${testID}_onBack`)}>
              <S.ArrowBackIcon />
            </TouchableOpacity>
          )}
        </S.Left>
        <S.Body>
          {title && <S.Title testID={`${testID}_title`}>{title}</S.Title>}
          {logo && <S.Logo testID={`${testID}_logo`} source={logoImage} />}
        </S.Body>
        <S.Right></S.Right>
      </S.ContainerTop>
      {subtitle && (
        <S.ContainerSubtitle>
          <Column flex={8} justifyContent='center'>
            {subtitle && <S.SubTitle testID={`${testID}_subtitle`}>{subtitle}</S.SubTitle>}
          </Column>
          <Column flex={1} justifyContent='center' alignItems='flex-end'>
            {onFilter && (
              <TouchableOpacity onPress={onFilter} {...assignTestId('TouchableOpacity', `${testID}_onFilter`)}>
                <S.FilterIcon />
              </TouchableOpacity>
            )}
          </Column>
        </S.ContainerSubtitle>
      )}
    </S.Container>
  )
}
