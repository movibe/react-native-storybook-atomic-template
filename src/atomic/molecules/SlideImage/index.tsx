import React, { useState } from 'react'
import { ImageSourcePropType, View } from 'react-native'
import Carousel, { Pagination } from 'react-native-snap-carousel'
import styled from 'styled-components/native'

import { Column } from '../../../atomic/atoms/Flex'
import { Image } from '../../../atomic/atoms/Image'
import { assignTestId } from '../../../utils'

export interface ISlideImageProps {
  testID?: string
  images: Array<ImageSourcePropType>
  width: number
  height?: number
  onPress?: () => void
  autoPlay?: boolean
  firstItem?: number
  showPagination?: boolean
}

const Container = styled(Column)`
  background-color: ${({ theme }) => theme.colors.background};
  border-bottom-width: 1px;
  border-color: ${({ theme }) => theme.colors.disabled};
  width: 100%;
`
export const SlideImage: React.FC<ISlideImageProps> = ({
  testID = 'SlideImage',
  images,
  width,
  onPress,
  height,
  firstItem,
  autoPlay,
  showPagination,
}): JSX.Element => {
  const [activeSlide, setActiveSlide] = useState(firstItem || 0)

  const renderItem = ({ item, index }) => (
    <Column testID={`${testID}_slide_item`} onPress={onPress} key={index}>
      <Image testID={`${testID}_slide_item_image`} source={{ ...item, height: height, width: width }} />
    </Column>
  )

  const onChangeSlide = (index) => {
    setActiveSlide(index)
  }

  return (
    <Container alignItems='center' testID={testID}>
      <Carousel
        {...assignTestId('View', `${testID}_slide`)}
        data={images}
        renderItem={renderItem}
        sliderWidth={width}
        itemWidth={width}
        firstItem={firstItem}
        onSnapToItem={onChangeSlide}
        autoplay={autoPlay}
        activeSlideAlignment='center'
      />
      {showPagination && (
        <View {...assignTestId('View', `SlideImage_pagination`)}>
          <Pagination
            dotsLength={images.length}
            activeDotIndex={activeSlide}
            dotStyle={{
              width: 10,
              height: 10,
              borderRadius: 5,
              backgroundColor: '#828282',
            }}
            inactiveDotStyle={{
              backgroundColor: '#D8D8D8',
            }}
            inactiveDotOpacity={0.4}
            inactiveDotScale={0.6}
          />
        </View>
      )}
    </Container>
  )
}
