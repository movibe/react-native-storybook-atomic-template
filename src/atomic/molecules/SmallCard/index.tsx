import * as React from 'react'
import { ImageSourcePropType, TouchableOpacityProps } from 'react-native'

import { Text as NText } from '../../../atomic/atoms/Text'
import styled from 'styled-components/native'
import { assignTestId } from '../../../utils'

const Container = styled.TouchableOpacity`
  box-shadow: ${(props) => props.theme.boxShadow};
  border-radius: ${(props) => props.theme.borderRadius};
  overflow: hidden;
`

const Background = styled.ImageBackground`
  flex: 1;
  overflow: hidden;
  justify-content: flex-end;
`

const Overlay = styled.View`
  flex: 0.5;
  justify-content: flex-end;
  background-color: rgba(0, 0, 0, 0.5);
`

const Text = styled(NText)`
  color: #fff;
  font-size: 14px;
  font-weight: bold;
  margin: 10px;
`

export interface ISmallCardProps extends TouchableOpacityProps {
  testID?: string
  text?: string
  image: ImageSourcePropType
  gradient?: boolean
  onPress?: () => void
  width?: number
  height?: number
}

export const SmallCard: React.FC<ISmallCardProps> = ({
  testID = 'SmallCard',
  text,
  image,
  gradient,
  width = 120,
  height = 126,
  ...props
}): JSX.Element => {
  return (
    <Container {...assignTestId('TouchableOpacity', testID)} {...props} style={{ width, height }}>
      <Background source={image}>
        {gradient ? (
          <Overlay>
            <Text testID={testID}>{text}</Text>
          </Overlay>
        ) : (
          <Text testID={testID}>{text}</Text>
        )}
      </Background>
    </Container>
  )
}
