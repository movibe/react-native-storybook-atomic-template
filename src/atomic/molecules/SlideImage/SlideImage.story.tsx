import { action } from '@storybook/addon-actions'
import { boolean, number } from '@storybook/addon-knobs'
import * as React from 'react'

import { SlideImage } from '../../../atomic/molecules/SlideImage'
import { faker } from '../../../utils'

export default {
  title: 'Molecules/SlideImage',
}

export const normal = () => (
  <SlideImage
    onPress={action('onPress')}
    images={[
      {
        uri: faker.image.cats(),
      },
      {
        uri: faker.image.imageUrl(),
      },
      {
        uri: faker.image.city(),
      },
    ]}
    width={250}
    height={440}
    autoPlay={boolean('autoPlay', true)}
    firstItem={number('firstItem', 0)}
    showPagination={boolean('showPagination', false)}
  />
)

export const custom = () => {
  ; <SlideImage
    images={[
      {
        uri: faker.image.people(),
      },
      {
        uri: faker.image.nature(),
      },
      {
        uri: faker.image.nightlife(),
      },
    ]}
    width={250}
    height={440}
  />
}
