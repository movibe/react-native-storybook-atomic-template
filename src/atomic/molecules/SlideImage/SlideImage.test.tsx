import * as React from 'react'
import { fireEvent, render } from 'react-native-testing-library'

import { ISlideImageProps, SlideImage } from '../../../atomic/molecules/SlideImage'
import { mockTestID, StorybookHoc } from '../../../utils'

describe('Molecules/SlideImage', () => {
  let props: ISlideImageProps = {
    testID: 'SlideImage',
    width: 200,
    height: 200,
    images: [{ uri: 'http://google.com' }, { uri: 'http://google.com' }, { uri: 'http://google.com' }],
  }
  const { toJSON, findByTestId, getAllByTestId, update } = render(<SlideImage {...props} />, {
    wrapper: StorybookHoc,
  })

  const container = mockTestID('View', props.testID!).testID
  it(`render container #${container}`, async () => {
    const sut = await findByTestId(container)
    expect(sut).toBeTruthy()
  })

  const carousel = mockTestID('View', `${props.testID!}_slide`).testID
  it(`render carousel #${carousel}`, async () => {
    const sut = await findByTestId(carousel)
    expect(sut).toBeTruthy()
  })

  it(`render carousel with firtsIndex #${carousel}`, async () => {
    props.firstItem = 1
    update(<SlideImage {...props} />)
    const sut = await findByTestId(carousel)
    expect(sut).toBeTruthy()
  })

  const pagination = mockTestID('View', `${props.testID!}_pagination`).testID
  it(`render pagination #${pagination}`, async () => {
    props.showPagination = true
    update(<SlideImage {...props} />)
    const sut = await findByTestId(pagination)
    expect(sut).toBeTruthy()
    expect(sut.children.length).toEqual(1)
  })

  const item = mockTestID('View', `${props.testID!}_slide_item`).testID
  it(`render item #${item}`, async () => {
    const sut = await getAllByTestId(item)
    expect(sut).toHaveLength(props.images.length)
  })

  const itemPress = mockTestID('TouchableOpacity', `${props.testID!}_slide_item`).testID
  it(`press item #${itemPress}`, async () => {
    props.onPress = jest.fn()
    update(<SlideImage {...props} />)
    const sut = await getAllByTestId(itemPress)
    fireEvent(sut[0], 'press')
    expect(props.onPress).toHaveBeenCalledTimes(1)
  })

  const image = mockTestID('Image', `${props.testID!}_slide_item_image`).testID
  it(`render image #${image}`, async () => {
    const sut = await getAllByTestId(image)
    expect(sut).toHaveLength(props.images.length)
  })
})
