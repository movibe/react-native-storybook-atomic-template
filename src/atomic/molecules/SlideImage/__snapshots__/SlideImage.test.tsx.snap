// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/SlideImage should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100vh",
      },
    ]
  }
>
  <View
    accessible={false}
    style={
      Array [
        undefined,
        Object {
          "alignItems": undefined,
          "borderColor": undefined,
          "borderWidth": 0,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": undefined,
          "padding": 10,
        },
      ]
    }
    testID="View_View"
  >
    <View
      accessible={false}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 1,
              "borderColor": "#f1f1f1",
              "width": "100%",
            },
          ],
          Object {
            "alignItems": "center",
            "borderColor": undefined,
            "borderWidth": 0,
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": undefined,
            "padding": 0,
          },
        ]
      }
      testID="View_SlideImage"
    >
      <RCTScrollView
        accessible={false}
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
        data={
          Array [
            Object {
              "uri": "http://google.com",
            },
            Object {
              "uri": "http://google.com",
            },
            Object {
              "uri": "http://google.com",
            },
          ]
        }
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={1}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={6}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={200}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={13}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onSnapToItem={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={200}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "width": 200,
          }
        }
        swipeThreshold={20}
        testID="View_SlideImage_slide"
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={13}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 200,
                }
              }
            >
              <View
                accessible={false}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": undefined,
                    "borderColor": undefined,
                    "borderWidth": 0,
                    "flex": undefined,
                    "flexDirection": "column",
                    "justifyContent": undefined,
                    "opacity": 1,
                    "padding": 0,
                  }
                }
                testID="TouchableOpacity_SlideImage_slide_item"
              >
                <Image
                  accessible={false}
                  source={
                    Object {
                      "height": 200,
                      "uri": "http://google.com",
                      "width": 200,
                    }
                  }
                  testID="Image_SlideImage_slide_item_image"
                />
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 200,
                }
              }
            >
              <View
                accessible={false}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": undefined,
                    "borderColor": undefined,
                    "borderWidth": 0,
                    "flex": undefined,
                    "flexDirection": "column",
                    "justifyContent": undefined,
                    "opacity": 1,
                    "padding": 0,
                  }
                }
                testID="TouchableOpacity_SlideImage_slide_item"
              >
                <Image
                  accessible={false}
                  source={
                    Object {
                      "height": 200,
                      "uri": "http://google.com",
                      "width": 200,
                    }
                  }
                  testID="Image_SlideImage_slide_item_image"
                />
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 200,
                }
              }
            >
              <View
                accessible={false}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": undefined,
                    "borderColor": undefined,
                    "borderWidth": 0,
                    "flex": undefined,
                    "flexDirection": "column",
                    "justifyContent": undefined,
                    "opacity": 1,
                    "padding": 0,
                  }
                }
                testID="TouchableOpacity_SlideImage_slide_item"
              >
                <Image
                  accessible={false}
                  source={
                    Object {
                      "height": 200,
                      "uri": "http://google.com",
                      "width": 200,
                    }
                  }
                  testID="Image_SlideImage_slide_item_image"
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={false}
        testID="View_SlideImage_pagination"
      >
        <View
          accessible={false}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 20,
                "paddingVertical": 30,
              },
              Object {
                "flexDirection": "row",
              },
              Object {},
            ]
          }
        >
          <View
            accessible={false}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#828282",
                  "borderRadius": 5,
                  "height": 10,
                  "opacity": 0.4072,
                  "transform": Array [
                    Object {
                      "scale": 0.6004732522277224,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
          <View
            accessible={false}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#D8D8D8",
                  "borderRadius": 5,
                  "height": 10,
                  "opacity": 0.4,
                  "transform": Array [
                    Object {
                      "scale": 0.6,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
          <View
            accessible={false}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#D8D8D8",
                  "borderRadius": 5,
                  "height": 10,
                  "opacity": 0.4,
                  "transform": Array [
                    Object {
                      "scale": 0.6,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
